packages/backend-core/src/db/lucene.ts

```diff
@@ -282,8 +282,7 @@ export class QueryBuilder<T> {
     }
 
     const equal = (key: string, value: any) => {
-      // 0 evaluates to false, which means we would return all rows if we don't check it
-      if (!value && value !== 0) {
+      if (value === null || value === undefined) {
```

Comment:
The `!value` here likely also includes `""` which we really don't want to allow through - its an empty state which isn't super well defined, but the API has always treated it as ignored so we should hold this; it might be helpful to have a function which checks if its an empty state, but not zero or false.